engine/storage/src/org/apache/cloudstack/storage/datastore/ObjectInDataStoreManagerImpl.java

```diff
@@ -171,7 +171,10 @@ public DataObject create(DataObject obj, DataStore dataStore) {
                     ss.setVolumeId(snapshot.getVolumeId());
                     SnapshotDataStoreVO snapshotDataStoreVO = snapshotDataStoreDao.findParent(dataStore.getRole(), dataStore.getId(), snapshot.getVolumeId());
                     if (snapshotDataStoreVO != null) {
-                        ss.setParentSnapshotId(snapshotDataStoreVO.getSnapshotId());
```

Comment:
@niteshsarda findParent method is not giving the correct result. Lets fix that rather than taking a decision later. Also we should not hardcode it as vmware, better to use hypervisorType enum. Differential snapshots are only applicable for XenServer, should use that not vmware.